pkg/cluster/prepare.go

```diff
@@ -78,12 +94,38 @@ func (p *Preparer) Prepare() error {
 			fmt.Printf("failed to remove temporary folder at location: %s. Please clean manually.", p.Location)
 		}
 	}()
+	if err := p.PreFlightCheck(); err != nil {
+		return err
+	}
 	if err := p.Fetcher.Fetch(context.Background(), p.BaseURL, p.Version, p.Location); err != nil {
 		return err
 	}
 	return p.Applier.Apply(p.Location, p.KubeContext, p.KubeConfig, p.DryRun)
 }
 
+// PreFlightCheck checks whether prepare can run or not.
+func (p *Preparer) PreFlightCheck() error {
+	fmt.Print("Checking if flux namespace exists...")
+	args := []string{"get", "namespace", p.FluxNamespace, "--output", "name"}
+	if output, err := p.Runner.Run(kubectlCmd, args...); err != nil {
+		fmt.Println("\nOutput from kubectl command: ", string(output))
+		return fmt.Errorf("failed to get flux namespace: %w", err)
+	}
+	fmt.Println("done.")
+	fmt.Print("Checking for flux CRDs...")
+	// check if flux is installed
+	for _, crd := range FluxCRDs {
+		args = []string{"get", "crd", crd, "--output", "name"}
+		if output, err := p.Runner.Run(kubectlCmd, args...); err != nil {
+			fmt.Println("\nOutput from kubectl command: ", string(output))
+			return fmt.Errorf("failed to get crd %s : %w", crd, err)
+		}
+	}
+
+	fmt.Println("done.")
+	return nil
+}
+
```

Comment:
do we want to error if flux isn't installed, or just log a warning/hint to the user? I think it would be nice if we didn't introduce a dependency ordering for somebody setting a cluster up. Letting the user know they've installed profiles but it won't work until flux is installed might be a better UX. What do you think?

Reply:
Agreed. We can output a warning on it. But warnings are a bit devious that they tend to be ignored and then the people have no idea what happened? Maybe we could add a flag to explicit bypass the pre-flight check error? Or something like... `--no-error-on-preflight` or something like that?

Reply:
> Maybe we could add a flag to explicit bypass the pre-flight check error? Or something like... `--no-error-on-preflight` or something like that?

That sounds like a reasonable middle ground :smile:

Reply:
@aclevername Done. WDYT?